samza-core/src/main/java/org/apache/samza/container/grouper/task/GroupByContainerCount.java

```diff
@@ -74,7 +74,7 @@ public GroupByContainerCount(int containerCount) {
     // Convert to a Set of ContainerModel
     Set<ContainerModel> containerModels = new HashSet<>();
     for (int i = 0; i < containerCount; i++) {
-      containerModels.add(new ContainerModel(i, taskGroups[i]));
+      containerModels.add(new ContainerModel(String.valueOf(i), taskGroups[i]));
```

Comment:
What kind of validation would you suggest? In the Grouper itself?
Yes, if the class expects all container/processor ids to be parsable as integers, it should have an explicit validation of all external processor ids with a message explaining that the grouper will not work with arbitrary strings like UUIDs. Then if someone mistakenly changes the code, they get a clear exception rather than strange grouping behavior.